jina/drivers/reduce.py

```diff
@@ -82,26 +82,30 @@ class ReduceAllDriver(ReduceDriver):
         It uses the last request as a reference.
     """
 
-    def _apply(self, doc: 'jina_pb2.Document', context_doc: 'jina_pb2.Document', field: str, *args, **kwargs):
-        if doc.id not in self.doc_pointers:
-            self.doc_pointers[doc.id] = doc
+    def __init__(self, *args, **kwargs):
+        kwargs.setdefault('traversal_paths', ['c'])
+        super().__init__(use_tree_traversal=True, *args, **kwargs)
```

Comment:
same here, why do we choose to pass by named argument and another with this `setdefault`?

Reply:
see above.